src/reducers/console.js

```diff
@@ -73,14 +73,19 @@ export default function console(stateIn, {type, payload, meta}) {
       return initialState;
     case 'CONSOLE_INPUT_CHANGED':
       return state.set('currentInputValue', payload.value);
-    case 'CONSOLE_LOG_PRODUCED':
-      return state.setIn(
-        ['history', meta.key],
-        new ConsoleEntry({
-          value: payload.value,
-          evaluatedByCompiledProjectKey: payload.compiledProjectKey,
-        }),
-      );
+    case 'CONSOLE_LOG_BATCH_PRODUCED': {
+      let previousMap = state.get('history');
+      payload.actions.forEach((action) => {
+        previousMap = previousMap.set(
+          action.key,
+          new ConsoleEntry({
+            value: action.value,
+            evaluatedByCompiledProjectKey: action.compiledProjectKey,
+          }),
+        );
+      });
+      return state.set('history', previousMap);
```

Comment:
In terms of the nicest way to do this in Immutable, I think `state.update()` is always preferable in this situation, and I do think this is probably a case where Immutable’s `withMutations` would be appropriate. As I understand it [`withMutations` is primarily indicated](https://facebook.github.io/immutable-js/docs/#/OrderedMap/withMutations) in situations where you are making a bunch of discrete changes to an immutable data structure, and it allows you to avoid creating a ton of intermediate objects that are immediately thrown away. Seems ideally suited to this situation.